data/transactions/asset.go

```diff
@@ -72,44 +72,49 @@ type AssetFreezeTxnFields struct {
 	FreezeAccount basics.Address `codec:"fadd"`
 
 	// FreezeAsset is the asset ID being frozen or un-frozen.
-	FreezeAsset basics.AssetID `codec:"faid"`
+	FreezeAsset basics.AssetIndex `codec:"faid"`
 
 	// AssetFrozen is the new frozen value.
 	AssetFrozen bool `codec:"afrz"`
 }
 
-func clone(m map[basics.AssetID]basics.AssetHolding) map[basics.AssetID]basics.AssetHolding {
-	res := make(map[basics.AssetID]basics.AssetHolding)
+func clone(m map[basics.AssetIndex]basics.AssetHolding) map[basics.AssetIndex]basics.AssetHolding {
+	res := make(map[basics.AssetIndex]basics.AssetHolding)
 	for id, val := range m {
 		res[id] = val
 	}
 	return res
 }
 
-func cloneParams(m map[uint64]basics.AssetParams) map[uint64]basics.AssetParams {
-	res := make(map[uint64]basics.AssetParams)
+func cloneParams(m map[basics.AssetIndex]basics.AssetParams) map[basics.AssetIndex]basics.AssetParams {
+	res := make(map[basics.AssetIndex]basics.AssetParams)
 	for id, val := range m {
 		res[id] = val
 	}
 	return res
 }
 
-func getParams(balances Balances, cid basics.AssetID) (basics.AssetParams, error) {
-	creator, err := balances.Get(cid.Creator, false)
+func getParams(balances Balances, aidx basics.AssetIndex) (basics.AssetParams, basics.Address, error) {
```

Comment:
Can you name the returned values here?  `AssetParams` was clear enough without a name, but figuring out this `Address` requires looking at the code of this function.

Reply:
done